layers/+lang/typescript/packages.el

```diff
@@ -19,6 +20,17 @@
         web-mode
         ))
 
+(unless (or (configuration-layer/layer-usedp 'javascript)
+            (configuration-layer/layer-usedp 'html))
+  (defun typescript/init-add-node-modules-path ()
+    (use-package add-node-modules-path :defer t)))
```

Comment:
We should probably not rely on this behaviour. I think a better idea would be to make a layer called something like `web-common` that owns this package, and the javascript and html layers can depend on it.

It's also not necessary to guard your `add-hook` calls like that.